### ISSUE TITLE
Simulate proper air resistance?

### ISSUE THREAD

**Issue by peter1745** [jrouwe/JoltPhysics#414]
Hi, I was just looking around the Jolt API and I was wondering if there's any plans for adding air resistance natively to Jolt? I tried playing around with linear and angular damping, but that doesn't really give the same effect. I realize it's probably possible for me to implement it myself, either by forking Jolt or by just calculating and applying air resistance myself.

From memory PhysX and Bullet doesn't simulate air resistance natively either, and I'm curious why that is. I may end up giving it a go myself by forking Jolt, but yeah, just curious as to why it's not natively done, or if I'm just not understanding how it's supposed to be done 🙂

**Comment by jrouwe** [comment]
I don't think you need to fork Jolt in order to do this. In its simplest form you can write a loop to iterate over all active bodies and apply an impulse = c * v^2 * dt to the bodies (make sure they don't flip velocity). See: https://github.com/jrouwe/JoltPhysics/blob/d5cf4634292ad9a35421ffceb42a50faf3ce5562/Jolt/Physics/Body/Body.cpp#L244-L254 (note that you can call this function as well, but if you don't want buoyancy and only drag, you're wasting some cycles)
